zk_toolbox/crates/zk_inception/src/commands/chain/init.rs

```diff
@@ -67,8 +82,28 @@ pub async fn init(
 ) -> anyhow::Result<()> {
     copy_configs(shell, &ecosystem_config.link_to_code, &chain_config.configs)?;
 
+    let consensus_keys = generate_consensus_keys();
     let mut general_config = chain_config.get_general_config()?;
+    let genesis_spec = Some(get_genesis_specs(chain_config, &consensus_keys));
+    let api_config = general_config
+        .api_config
+        .clone()
+        .context(MSG_API_CONFIG_MISSING_ERR)?;
+    let public_addr = parse_public_addr(&api_config)?;
```

Comment:
you are getting api config and they applying offset, seems you have to do in another order